### ISSUE TITLE
The caret is not moved when typing a decimal separator at an existing decimal separator

### ISSUE THREAD

**Issue by pvdstel** [s-yadav/react-number-format#711]
#### Describe the issue and the actual behavior

Typing one of the characters in `allowedDecimals` does not update the caret position, if the caret is in front of an already-existing decimal separator.

This is especially noticeable if `fixedDecimalScale` is enabled, since it forces a decimal to be present.

This error started occurring in `5.1.0`.

#### Describe the expected behavior

If a decimal symbol is typed (any of the allowed decimals) at the same position as an existing decimal sign, the caret should move to the right of the decimal sign, as if the decimal separator was "overwritten" with a new one.

#### Provide a CodeSandbox link illustrating the issue

https://codesandbox.io/s/alloweddecimalseparators-fixeddecimalscale-bug-epp4d1?file=/package.json

#### Provide steps to reproduce this issue

Type something like <kbd>12%34</kbd> in the input field and notice that the decimal sign was never typed.

If the version of `react-number-format` is downgraded to `5.0.1` it works as expected.

#### Please check the browsers where the issue is seen
- [x] Chrome
- [ ] Chrome (Android)
- [ ] Safari (OSX)
- [ ] Safari (iOS)
- [x] Firefox
- [ ] Firefox (Android)

**Comment by jarekpelczynski** [comment]
Have the same issue

If I have provided these options, pressing dot (.) do nothing
```
decimalSeparator=","
allowedDecimalSeparators={[".", ","]}
decimalScale={2}
```

**Comment by s-yadav** [comment]
Will check this.

**Comment by adrian-amigues** [comment]
I confirm having this issue as well, although I notice that the bug seems only present when `fixedDecimalScale` is `true`

**Comment by s-yadav** [comment]
Fix is available on 5.1.3
